packages/next/build/webpack/loaders/next-image-loader.js

```diff
@@ -0,0 +1,58 @@
+import loaderUtils from 'next/dist/compiled/loader-utils'
+import sizeOf from 'image-size'
+import { processBuffer } from '../../../next-server/server/lib/squoosh/main'
+import fs from 'fs'
+
+async function nextImageLoader(content) {
+  this.cacheable && this.cacheable(true)
+  this.addDependency(this.resourcePath)
+
+  const context = this.rootContext
+  const opts = { context, content }
+  const interpolatedName = loaderUtils.interpolateName(
+    this,
+    '/[path][name].[ext]',
+    opts
+  )
+
+  let extension = loaderUtils.interpolateName(this, '[ext]', opts)
+  if (extension === 'jpg') {
+    extension = 'jpeg'
+  }
+
+  if (interpolatedName.slice(0, 7) !== '/public') {
+    const err = new Error(
+      'Static Image loader used with filepath not in the /public directory: ' +
+        interpolatedName
+    )
+    this.emitError(err)
+    return
+  }
+
+  const src = interpolatedName.slice(7)
+  const imageSize = sizeOf(this.resourcePath)
+  let dataURI
+  if (extension === 'jpeg' || extension === 'png') {
+    const fileBuffer = Buffer.from(fs.readFileSync(this.resourcePath))
+    const resizedImage = await processBuffer(
+      fileBuffer,
+      [{ type: 'resize', width: 6 }],
```

Comment:
I wonder if this is sub-optimal for images that are very high, but have a small width...

It might be better to resize it to width: 6 when width > height, otherwise to height: 6 (for width <= height)

Reply:
Done